tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java

```diff
@@ -114,92 +119,91 @@ public GitHubTask(boolean dryRunMode) {
+            case COMMENTPR -> commentPR(in, apiInfo);
+            case GETPRCOMMITLIST -> getPRCommitList(in, apiInfo);
+            case MERGEPR -> mergePR(in, apiInfo);
+            case CLOSEPR -> closePR(in, apiInfo);
+            case CREATEISSUE -> createIssue(in, apiInfo);
+            case CREATETAG -> createTag(in, apiInfo);
+            case DELETETAG -> deleteTag(in, apiInfo);
+            case DELETEBRANCH -> deleteBranch(in, apiInfo);
+            case MERGE -> merge(in, apiInfo);
+            case GETCOMMIT -> getCommit(in, apiInfo);
+            case ADDSTATUS -> addStatus(in, apiInfo);
+            case GETSTATUSES -> getStatuses(in, apiInfo);
+            case FORKREPO -> forkRepo(in, apiInfo);
+            case GETBRANCHLIST -> getBranchList(in, apiInfo);
+            case GETPR -> getPR(in, apiInfo);
+            case GETPRLIST -> getPRList(in, apiInfo);
+            case GETTAGLIST -> getTagList(in, apiInfo);
+            case GETLATESTSHA -> getLatestSHA(in, apiInfo);
+            case CREATEREPO -> createRepo(in, apiInfo);
+            case DELETEREPO -> deleteRepo(in, apiInfo);
+            case GETCONTENT -> getContent(in, apiInfo);
+            case CREATEHOOK -> createHook(in, apiInfo);
+            case GETPRFILES -> getPRFiles(in, apiInfo);
+            case CREATEAPPTOKEN -> createAppToken(apiInfo);
+        };
+    }
+
+    protected AccessTokenProvider getTokenProvider(Map<String, Object> in, String baseUrl, GitSecretService secretService) {
+        String owner = assertString(in, GITHUB_ORGNAME);
+        String repo = assertString(in, GITHUB_REPONAME);
+
+        Map<String, Map<String, Object>> auth = getMap(in, "auth", Map.of());
+        String token = getString(in, GITHUB_AUTH_ACCESSTOKEN, null);
+
+        if (auth.isEmpty()) {
+            log.warn("Using deprecated 'accessToken' input. Please use 'auth.accessToken.token' instead.");
+            // maintain support for old top-level input, translate to new "auth.accessToken" input
+            auth = Map.of(GITHUB_AUTH_ACCESSTOKEN, Map.of("token", token));
+        }
+
+        if (auth.size() != 1) {
+            throw new IllegalArgumentException("Invalid 'auth' input. Expected one element, got: " + auth.keySet());
         }
+
+        Map.Entry<String, Map<String, Object>> authType = auth.entrySet().stream().findFirst()
+                .orElseThrow(() -> new IllegalStateException("Invalid 'auth' input. Missing 'auth' entry"));
+
+        Class<? extends Auth> authClass = switch(authType.getKey()) {
+            case GITHUB_AUTH_ACCESSTOKEN -> Auth.AccessTokenAuth.class;
+            case GITHUB_AUTH_APP_INSTALLATION -> Auth.AppInstallationAuth.class;
+            case GITHUB_AUTH_APP_INSTALLATION_SECRET -> Auth.AppInstallationSecretAuth.class;
+            default -> throw new IllegalArgumentException("Unsupported auth type: " + authType.getKey());
+        };
+
+        Auth a = Utils.getObjectMapper().convertValue(authType.getValue(), authClass);
```

Comment:
There is `private static final ObjectMapper objectMapper` already?
Also, GitHubTaskV2 should use @Inject-able ObjectMapper (provided by https://github.com/walmartlabs/concord/blob/master/runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/guice/ObjectMapperProvider.java)

Reply:
There's definitely too many `ObjectMapper`'s being instantitated. Easy to clean that up.

Regarding injection, easy enough to do for runtime-v2, then throw in a method to whip one up in runtime-v2. However, I have a concern that there's no protection from some other task going ham enabling or disabling features causing side effects. Say, changing timestamp handling, or adding custom deserializers. The _should_ [copy it](https://github.com/walmartlabs/concord-plugins/blob/739608a0dea7f7317a8ae494c05c09c5eb479b39/tasks/aws/src/main/java/com/walmartlabs/concord/plugins/aws/EcrTask.java#L53), but it seem pretty easy for accidents to happen.